### PR TITLE
Stop Generating 2 Player Sessions

### LINKED ISSUE #431: Connecting to GameLift Using Json Menu Ui Creates 2nd Player Session
Connecting the GameLift via the GameLiftConnectJsonMenu level will cause a 2nd player-session to get created

**Steps to reproduce**
1. Run a server on GameLift and connect a client (follow the instructions here if needed https://github.com/o3de/o3de-multiplayersample/blob/development/MPSGameLift/Documentation/GameLift.md)
2. After connecting open the GameLift dashboard > Fleets > Game Sessions > Player Sessions

**Expected behavior**
Game session contains 1 player session.
The developer has already created the player session, the game should use that one and not create another.

**Actual behavior**
The game generates and uses a 2nd player session
Notice the 2nd player-session is using the 1st player-session's id as the player id

image

**Found in Branch**
development
June 08 2023

### PR DESCRIPTION
**Pull request by AMZN-Gene**
Fix to properly use the player session data. 
Before this change the menu was using a AWS method to create a new player session... in this case, the developer already has the player session

Fixes #431

### CODE REVIEW

File: MPSGameLift/Code/Source/Components/UI/UiGameLiftConnectWithPlayerSessionData.cpp [hunk 1]
```diff
@@ -110,35 +109,51 @@ namespace MPSGameLift
...
             return;
         }
 
-        if (!document.HasMember("PlayerSessionId"))
+        if (!document.HasMember("IpAddress") && !document.HasMember("DnsName"))
         {
-            UiTextBus::Event(m_jsonParseFailTextUi, &UiTextInterface::SetText, "Missing PlayerSessionId!");
+            UiTextBus::Event(m_jsonParseFailTextUi, &UiTextInterface::SetText, "Must provide either an IpAddress or DnsName!");
             return;
         }
```
Comment (AMZN-alexpete) [inline MPSGameLift/Code/Source/Components/UI/UiGameLiftConnectWithPlayerSessionData.cpp:123]: is it not easier to connect via game session id?  won't you not know the server IP or DNS name until the fleet is initialized or something and you look for that info in the console?

Comment (AMZN-Gene) [reply MPSGameLift/Code/Source/Components/UI/UiGameLiftConnectWithPlayerSessionData.cpp:123]: Easier, yes. There’s a AWS GameLift gem API to connect by passing a game session id. That method will ask game lift to give back a player session, dns, and port and then does what my function is doing behind the scenes. For the 5-Click-Sample the developer will have generated the player session id themselves using the GameLift website.

**Comment by AMZN-daimini** [comment]
Ooh this explains things. During my testing, if I would set the player limit on a server to 1, it would not let me connect immediately. I would have to wait for the user-generated player session to expire, then try again and then it would connect.

So if anything, it looks like that the current setup did not have the 60-second expiration limitation for connecting via the client, as the client was generating a new player session anyways.

File: MPSGameLift/Code/Source/Components/UI/UiGameLiftConnectWithPlayerSessionData.cpp [hunk 2]
```diff
@@ -95,12 +94,12 @@ namespace MPSGameLift
 
         if (gameLiftJsonString.empty())
         {
-            UiTextBus::Event(m_jsonParseFailTextUi, &UiTextInterface::SetText, "Please provide GameLift GameSessionId and PlayerSessionId in JSON format!");
+            UiTextBus::Event(m_jsonParseFailTextUi, &UiTextInterface::SetText, "Please provide GameLift player connection information in JSON format!");
```
[the remark this answers is not in the document]
Comment (AMZN-Gene) [reply MPSGameLift/Code/Source/Components/UI/UiGameLiftConnectWithPlayerSessionData.cpp:97]: GameLift clients use the ip, port, and player session id. Without the player session id, the server will reject the incoming connection.
Technically we only need those 3 pieces of data, but it's must easier to just paste in the JSON block wholesale.

**Review by AMZN-daimini — APPROVED** [review]

**Review by AMZN-alexpete — APPROVED** [review]

**Pull request merged**